### PR TITLE
Enable to skip JUNK chunk when reading a WAV

### PR DESCRIPTION
**Pull request by kompadre**
Enables to skip the "JUNK" section when reading a WAV sample (https://www.daubnet.com/en/file-format-riff).

This extra chunk before "fmt" and that should be skipped appears in samples ripped from CDs: apparently CD-player can only move in certain increments and data that should be skipped makes sure that the start of audio data aligns with what player expects. Is this chunk useless nowdays? - absolutely. Will it help to be able to load old samples? - i think it will. As mentioned before half of my goto samples library cant be loaded because of this.

### CODE REVIEW

**Comment by djdiskmachine** [comment]
Thanks for sticking with it!
I'll check it out closer later this weekend, cheers.

**Comment by kompadre** [comment]
Never lost my faith.

**Review by djdiskmachine — COMMENTED** [review]
Looks a lot leaner than the last one, are we still refusing 0kHz sample rates and the like?

File: sources/Application/Instruments/WavFile.cpp [hunk 1]
```diff
@@ -117,13 +117,30 @@ WavFile *WavFile::Open(const char *path) {
...
-	if (chunk!=0x20746D66) {
+
+        // Read (possible) JUNK
+
+	#ifndef NOSKIPJUNK
+	if (chunk==0x4b4e554a) {
+		position+=wav->readBlock(position,4) ;
+        memcpy(&size, wav->readBuffer_,4) ;
+        size = Swap32(size) ;
+        // Trace::Debug("WavFile::Open(): skipping JUNK with size=%d", size);
```
[the remark this answers is not in the document]
Comment (djdiskmachine) [reply sources/Application/Instruments/WavFile.cpp:133]: As in uncomment please as its useful and also no commented-out code please.

**Comment by kompadre** [comment]
Yes, definitely. No 0khz shall pass. The only thing this change does is: read if there's a skippable chunk, skip those bytes and read the next chunk. My theory about 0 defaults turned to be unproven

**Comment by kompadre** [comment]
Is something's still amiss?

**Comment by djdiskmachine** [comment]
> Is something's still amiss?

Sorry for the delay. I haven't gotten the time to check this out proper yet, hopefully next week. Looks fine but I want to try it out :)

**Review by djdiskmachine — APPROVED** [review]

**Pull request merged**